### PR TITLE
feat: refactor and make script runnable inside kubernetes

### PR DESCRIPTION
**Pull request by artsyjian**
Refactors:

- Move much of lib code to main program and delete unused libs.
- Delete vault lib's login wrapper. Let client call login methods directly.
- Parameterize service account token path.

Tested working using sanbox app with the following updates to its Hokusai spec:
```
ip-10-0-0-17:hokusai-sandbox jxu$ git diff main
diff --git a/hokusai/staging.yml b/hokusai/staging.yml
index ab6959a..e9c76f1 100644
--- a/hokusai/staging.yml
+++ b/hokusai/staging.yml
@@ -32,11 +32,12 @@ spec:
     spec:
       initContainers:
       - name: setenv
-        image: 585031190124.dkr.ecr.us-east-1.amazonaws.com/opstools:staging
+        image: 585031190124.dkr.ecr.us-east-1.amazonaws.com/fortress:test
         imagePullPolicy: Always
         command:
         - python
-        - src/load_secrets_from_vault/load.py
+        - src/load/load.py
+        - kubernetes
         - staging
         - hokusai-sandbox
         envFrom:
```

### CODE REVIEW

File: src/lib/vault.py [hunk 1]
```diff
@@ -4,60 +4,44 @@
...
     self._mount_point = kvv2_mount_point
     self._path = path
-    self._login(auth_method, token, role)
-
-  def _login(self, auth_method, token=None, role=None):
-    ''' log into Vault using the specified method '''
-    if auth_method == 'iam':
-      self._iam_login(role)
-    elif auth_method == 'token':
-      self._client.token = token
```
Comment (artsyjian) [inline src/lib/vault.py:31]: token login is insecure. let's not even allow it.

File: src/lib/vault.py [hunk 2]
```diff
@@ -4,60 +4,44 @@
...
-        credentials.access_key,
-        credentials.secret_key,
-        credentials.token,
-        role=role
+    # if role is None, hvac defaults it to iam username
+    self._client.auth.aws.iam_login(
+      credentials.access_key,
+      credentials.secret_key,
+      credentials.token,
+      role=role
```
Comment (artsyjian) [inline src/lib/vault.py:31]: ok to pass role even if it's None

**Review by cavvia — COMMENTED** [review]
One non-blocking comment... leaving this open for a bit in case you feel like acting on it, otherwise can merge.

File: src/load/load/load.py [hunk 3]
```diff
@@ -4,26 +4,28 @@
...
 import load.context
 
-from lib.util import (
-  url_host_port
-)
 from lib.vault import Vault
 
 
-def load_secrets(project, vault_host, vault_port, auth, role, secrets_file, kvv2_mount_point):
+def load_secrets(project, vault_host, vault_port, auth, role, sa_token_path, secrets_file, kvv2_mount_point):
```
[the remark this answers is not in the document]
Comment (artsyjian) [reply src/load/load/load.py:10]: yeah, i've been un-comfrotable about such many positional args.

[the remark this answers is not in the document]
Comment (artsyjian) [reply src/load/load/load.py]: i like the idea of using a struct-like object to group related args. here's an [attempt](https://github.com/artsy/fortress/pull/6/commits/eb5cc304c9fbea87f4ab009bd69df545d08a5d41) using `dataclass`. can also use plain old dictionary, plain old class, or `namedtuple`, but dataclass seems most elegant.

[the remark this answers is not in the document]
Comment (artsyjian) [reply src/load/load/load.py]: in regards to `black`, we [tried](https://artsy.slack.com/archives/CA8SANW3W/p1681400140011869) it on Opstools repo, and was stalled at indent size.

[the remark this answers is not in the document]
Comment (artsyjian) [reply src/load/load/load.py]: let's [poll](https://artsy.slack.com/archives/CA8SANW3W/p1731433368423189) !

[the remark this answers is not in the document]
Comment (artsyjian) [reply src/load/load/load.py]: [added black](https://github.com/artsy/fortress/pull/6/commits/1348b7557b4f16a48aec57aa4c9b721b53865f7c).

**Review by cavvia — APPROVED** [review]
Looks better now with a DataClass to marshal the vault options.

**Pull request merged**